src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs

```diff
@@ -120,8 +113,8 @@ internal sealed partial class LightweightRenameLocations
 
             return new LightweightRenameLocations(
                 solution, options, fallbackOptions, renameLocations.Locations,
-                renameLocations.ImplicitLocations.IsDefault ? null : renameLocations.ImplicitLocations.Select(loc => SerializableReferenceLocation.Dehydrate(loc, cancellationToken)).ToArray(),
-                renameLocations.ReferencedSymbols.IsDefault ? null : renameLocations.ReferencedSymbols.Select(sym => SerializableSymbolAndProjectId.Dehydrate(solution, sym, cancellationToken)).ToArray());
+                renameLocations.ImplicitLocations.SelectAsArray(loc => SerializableReferenceLocation.Dehydrate(loc, cancellationToken)),
```

Comment:
Should these be `SelectAsArrayAsync`?

Reply:
no, these are fine dehydration in this case can be synchronous. (it's often just making simple keys from data)
rehydration is often the expensive step, having to take those keys and map to corresponding expensive objects.